pex/tools/commands/venv.py

```diff
@@ -0,0 +1,236 @@
+
+        venv = Virtualenv.create(options.venv[0], interpreter=pex.interpreter, force=options.force)
+
+        # 1. Populate the venv with the PEX contents.
+        pex_info = pex.pex_info()
+        if zipfile.is_zipfile(pex.path()):
+            PEXEnvironment.explode_code(
+                pex.path(), pex_info, venv.site_packages_dir, exclude=("__main__.py",)
+            )
+        else:
+            _copytree(
+                src=pex.path(),
+                dst=venv.site_packages_dir,
+                exclude=(pex_info.internal_cache, pex_builder.BOOTSTRAP_DIR, "__main__.py"),
+            )
+
+        for dist in pex.activate():
+            _copytree(
+                src=dist.location,
+                dst=venv.site_packages_dir,
+                exclude=("bin",),
+                collisions_ok=options.collisions_ok,
+            )
+            dist_bin_dir = os.path.join(dist.location, "bin")
+            if os.path.isdir(dist_bin_dir):
+                _copytree(dist_bin_dir, venv.bin_dir, collisions_ok=options.collisions_ok)
+
+        # 2. Add a __main__ to the root of the venv for running the venv dir like a loose PEX dir
+        # and a main.py for running as a script.
+        main_contents = dedent(
+            """\
+            #!{venv_python} -sE
+
+            import os
+            import sys
+
+            python = {venv_python!r}
+            if sys.executable != python:
+                os.execv(python, [python, "-sE"] + sys.argv)
+
+            os.environ["VIRTUAL_ENV"] = {venv_dir!r}
+            sys.path.extend(os.environ.get("PEX_EXTRA_SYS_PATH", "").split(os.pathsep))
+
+            bin_dir = {venv_bin_dir!r}
+            bin_path = {bin_path!r}
+            if bin_path:
+                PATH = os.environ.get("PATH", "").split(os.pathsep)
+                if bin_path == "prepend":
+                    PATH = [bin_dir] + PATH
+                else:
+                    PATH.append(bin_dir)
+                os.environ["PATH"] = os.pathsep.join(PATH)
+
+            PEX_OVERRIDE_KEYS = ("PEX_INTERPRETER", "PEX_SCRIPT", "PEX_MODULE")
+            pex_overrides = dict(
+                (key, os.environ.pop(key)) for key in PEX_OVERRIDE_KEYS if key in os.environ
+            )
+            if len(pex_overrides) > 1:
+                sys.stderr.write(
+                    "Can only specify one of {{overrides}}; found: {{found}}\\n".format(
+                        overrides=", ".join(PEX_OVERRIDE_KEYS),
+                        found=" ".join("{{}}={{}}".format(k, v) for k, v in pex_overrides.items())
+                    )
+                )
+                sys.exit(1)
+
+            pex_script = pex_overrides.get("PEX_SCRIPT")
+            if pex_script:
+                script_path = os.path.join(bin_dir, pex_script)
+                os.execv(script_path, [script_path] + sys.argv[1:])
+
+            # TODO(John Sirois): Support `-c`, `-m` and `-` special modes when PEX_INTERPRETER is
```

Comment:
TODOs with github issues are a bit easier to maintain and add information to.

Reply:
Yup, added.